include/deal.II/fe/fe_values.h

```diff
@@ -5429,6 +5507,43 @@ FEValuesBase<dim, spacedim>::get_inverse_jacobians() const
 
 
 
+template <int dim, int spacedim>
+inline boost::integer_range<unsigned int>
+FEValuesBase<dim, spacedim>::local_dof_indices() const
+{
+  return boost::irange(0U, dofs_per_cell);
+}
+
+
+
+template <int dim, int spacedim>
+inline boost::integer_range<unsigned int>
+FEValuesBase<dim, spacedim>::local_dof_indices_starting_at(
+  const unsigned int start_dof_index) const
+{
+  Assert(
+    start_dof_index <= dofs_per_cell,
+    ExcMessage(
+      "The start DoF index is beyond the number of degrees of freedom that the cell has."));
+  return boost::irange(start_dof_index, dofs_per_cell);
+}
+
+
+
+template <int dim, int spacedim>
+inline boost::integer_range<unsigned int>
+FEValuesBase<dim, spacedim>::local_dof_indices_ending_at(
+  const unsigned int end_dof_index) const
+{
+  Assert(
+    end_dof_index + 1 <= dofs_per_cell,
```

Comment:
So the issue here is that, [according to boost's documentation](https://www.boost.org/doc/libs/1_67_0/libs/range/doc/html/range/reference/ranges/irange.html), 
>  It should be noted that the first and last parameters denoted a half-open range. 

So the last entry cannot ever be greater than `dofs_per_cell`. And we wish to include `end_dof_index` during iteration, so I must set the `last` iterator to be `end_dof_index + 1` .

Reply:
Ah, I see -- in contrast to all of the other places you're asking for a closed range here. Then I guess this and the other place make sense. 

I did think whether it makes sense to change the function so that it asks for a half-open range, but I see value in making things symmetric: the special index given to the two functions is always included.